build/tasks/PackagePublisher.cs

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Build.Framework;
+using NuGet.Common;
+using NuGet.Packaging;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+using BuildTask = Microsoft.Build.Utilities.Task;
+
+namespace UniverseTools
+{
+    public class PackagePublisher : BuildTask, ICancelableTask
```

Comment:
Can we push this task in to the SDK? I really don't want us to maintain 4 copies of this (CoreCLR, Coherence, UniverseCoherence, Coherence-Signed).